wagtail/wagtailadmin/viewsets/__init__.py

```diff
@@ -0,0 +1,34 @@
+from django.conf.urls import url, include
+
+from wagtail.wagtailcore import hooks
+
+
+class ViewSetRegistry(object):
+    def __init__(self):
+        self.viewsets = []
+
+    def populate(self):
+        for fn in hooks.get_hooks('register_admin_viewset'):
+            viewset = fn()
+            self.register(viewset)
+
+    def register(self, viewset_cls):
+        self.viewsets.append(viewset_cls)
+        return viewset_cls
+
+    def get_urlpatterns(self):
+        urlpatterns = []
+
+        for viewset in self.viewsets:
+            vs_urlpatterns = viewset.get_urlpatterns()
+
+            if vs_urlpatterns:
+                urlpatterns.append(url(
+                    r'^{}/'.format(viewset.name),
```

Comment:
Using `viewset.name` as both the URL namespace and the URL path prefix means that in the case of wagtailsites, the URL has suddenly changed from `/admin/sites/` to `/admin/wagtailsites/`, which is something I think we should avoid.

I see `name` is assigned at the point that `ViewSet` is instantiated, in `register_admin_viewset` - presumably with the intention that different instances of the same viewset class could coexist in different URL namespaces - so it probably makes sense to add a new optional kwarg at that point:

    @hooks.register('register_admin_viewset')
    def register_viewset():
        return SiteViewSet('wagtailsites', url_prefix='sites')

with `url_prefix` defaulting to `self.name` if not specified.